src/Game.js

```diff
@@ -0,0 +1,85 @@
+class Game {
+    'use strict';
+    constructor(scorecard){
+      this.scorecard = this._scorecard = typeof scorecard !== 'undefined' ? scorecard : new ScoreCard();
+      this.frame = 1;
+      this.roll = 1;
+      this.knocked = 0;
+      this.isEnded = false;
+    }
+
+    play(pins){
```

Comment:
Play could maybe be broken down into more classes/methods to simplify the flow a little?